File: app/src/main/java/com/sagar/watchnext/network/models/tv/details/Details.java

package com.sagar.watchnext.network.models.tv.details;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Details {

    @SerializedName("backdrop_path")
    private String backdropPath;

    @SerializedName("created_by")
    private List<CreatedBy> createdByList;

    @SerializedName("episode_run_time")
    private List<Integer> episodeRunTimeList;

    @SerializedName("first_air_date")
    private String firstAirDate;

    @SerializedName("genres")
    private List<Genre> genres;

    @SerializedName("homepage")
    private String homepage;

    @SerializedName("id")
    private int id;

    @SerializedName("in_production")
    private boolean inProduction;

    @SerializedName("languages")
    private List<String> languages;

    @SerializedName("last_air_date")
    private String lastAirDate;

    @SerializedName("name")
    private String name;

    @SerializedName("networks")
    private List<Network> networks ;

    @SerializedName("number_of_episodes")
    private int numberOfEpisodes;

    @SerializedName("number_of_seasons")
    private int numberOfSeasons;

    @SerializedName("origin_country")
    private List<String> originCountries;

    @SerializedName("original_language")
    private String originalLanguage;

    @SerializedName("original_name")
    private String originalName;

    @SerializedName("overview")
    private String overview;

    @SerializedName("popularity")
    private float popularity;

    @SerializedName("poster_path")
    private String posterPath;

    @SerializedName("production_companies")
    private List<ProductionCompany> productionCompanies;

    @SerializedName("seasons")
    private List<Season> seasons;

    @SerializedName("status")
    private String status;

    @SerializedName("type")
    private String type;

    @SerializedName("vote_average")
    private int voteAverage;

    @SerializedName("vote_count")
    private int voteCount;

}

File: app/src/main/java/com/sagar/watchnext/network/models/tv/details/Details.kt
package com.sagar.watchnext.network.models.tv.details

import com.google.gson.annotations.SerializedName

class Details {

    @SerializedName("backdrop_path")
    val backdropPath: String? = null

    @SerializedName("created_by")
    val createdByList: List<CreatedBy>? = null

    @SerializedName("episode_run_time")
    val episodeRunTimeList: List<Int>? = null

    @SerializedName("first_air_date")
    val firstAirDate: String? = null

    @SerializedName("genres")
    val genres: List<Genre>? = null

    @SerializedName("homepage")
    val homepage: String? = null

    @SerializedName("id")
    val id: Int = 0

    @SerializedName("in_production")
    val isInProduction: Boolean = false

    @SerializedName("languages")
    val languages: List<String>? = null

    @SerializedName("last_air_date")
    val lastAirDate: String? = null

    @SerializedName("name")
    val name: String? = null

    @SerializedName("networks")
    val networks: List<Network>? = null

    @SerializedName("number_of_episodes")
    val numberOfEpisodes: Int = 0

    @SerializedName("number_of_seasons")
    val numberOfSeasons: Int = 0

    @SerializedName("origin_country")
    val originCountries: List<String>? = null

    @SerializedName("original_language")
    val originalLanguage: String? = null

    @SerializedName("original_name")
    val originalName: String? = null

    @SerializedName("overview")
    val overview: String? = null

    @SerializedName("popularity")
    val popularity: Float = 0.toFloat()

    @SerializedName("poster_path")
    val posterPath: String? = null

    @SerializedName("production_companies")
    val productionCompanies: List<ProductionCompany>? = null

    @SerializedName("seasons")
    val seasons: List<Season>? = null

    @SerializedName("status")
    val status: String? = null

    @SerializedName("type")
    val type: String? = null

    @SerializedName("vote_average")
    val voteAverage: Float = 0.toFloat()

    @SerializedName("vote_count")
    val voteCount: Int = 0
}
